apps/expo-go/ios/Exponent/Kernel/DevSupport/EXPackagerLogHelper.swift

```diff
@@ -0,0 +1,143 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+import Foundation
+import React
+
+@objc enum EXPackagerLogLevel: Int {
+  case info
+  case warning
+  case error
+}
+
+@objc class EXPackagerLogHelper: NSObject, SRWebSocketDelegate {
+  private var connected: Bool = false
+  private var socket: SRWebSocket?
+  private var bundleURL: URL
+  private var pendingMessage: String?
+  private var logLevel: EXPackagerLogLevel
+  private var onComplete: (() -> Void)?
+
+  @objc init(bundleURL: URL, level: EXPackagerLogLevel) {
+    self.bundleURL = bundleURL
+    self.logLevel = level
+  }
+
+  deinit {
+    socket?.close()
+    self.socket = nil
+    connected = false
+  }
+
+  @objc func sendMessage(_ message: String, withCompletion completion: @escaping () -> Void) {
+    pendingMessage = message
+    onComplete = completion
+    if !connected {
+      createSocket()
+    } else {
+      sendPendingMessage()
+    }
+  }
+
+  @objc static func logInfo(_ message: String, withBundleUrl url: URL) {
+    log(message, withBundleUrl: url, level: .info)
+  }
+
+  @objc static func logWarning(_ message: String, withBundleUrl url: URL) {
+    log(message, withBundleUrl: url, level: .warning)
+  }
+
+  @objc static func logError(_ message: String, withBundleUrl url: URL) {
+    log(message, withBundleUrl: url, level: .error)
+  }
+
+  @objc static func log(_ message: String, withBundleUrl url: URL, level: EXPackagerLogLevel) {
+    var strongHelper: EXPackagerLogHelper? = EXPackagerLogHelper(bundleURL: url, level: level)
+    strongHelper?.sendMessage(message) {
+      strongHelper = nil
+    }
+  }
+
+  func webSocketDidOpen(_ webSocket: SRWebSocket) {
+    connected = true
+    sendPendingMessage()
+  }
+
+  func webSocket(_ webSocket: SRWebSocket, didFailWithError error: Error) {
+    // Ignored
+  }
+
+  func webSocket(_ webSocket: SRWebSocket, didCloseWithCode code: Int, reason: String?, wasClean: Bool) {
+    socket = nil
+    connected = false
+  }
+
+  private func sendPendingMessage() {
+    guard let message = pendingMessage else {
+      return
+    }
+
+    guard let socket = socket else {
+      return
+    }
+
+    let type: String
+    switch logLevel {
+    case .info:
+      type = "info"
+    case .warning:
+      type = "warn"
+    case .error:
+      type = "error"
+    }
```

Comment:
We could make the EXPackagerLogLevel cases match these strings and drop this switch statement. Then return `"level":  logLevel.rawValue`,

Reply:
Good point. Updated!